pyrelational/model_managers/model_utils.py

```diff
@@ -1,23 +1,25 @@
 import warnings
-from typing import List, Union
+from typing import Any, Dict, List, Union
 
 import torch
 
 
-def _determine_device(gpus: Union[List[int], str, int, None]) -> torch.device:
+def _determine_device(trainer_config: Dict[str, Any]) -> torch.device:
     """
     Determines the torch device of the model from the gpus argument for pytorch lightning trainer
 
-    :param gpus: Number of gpus (int) or which gpus to train on (str, list)
+    :param trainer_config: configuration dictionary for a pytorch lightning Trainer
     :return: torch device
     """
-    if isinstance(gpus, list):
-        gpus = str(gpus[0])
-        warnings.warn("Multiple GPUs provided, setting the first GPU to be device used in call function of model")
-        return torch.device(f"cuda:{gpus}")
-    elif isinstance(gpus, str):
-        return torch.device(f"cuda:{gpus}")
-    elif isinstance(gpus, int) and (gpus > 0):
-        return torch.device("cuda")
-    else:
+    if trainer_config.get("accelerator", "cpu") == "cpu":
         return torch.device("cpu")
+    else:
+        devices: Union[List[int], str, int] = trainer_config.get("devices", "auto")
+        if isinstance(devices, str):
```

Comment:
Why is it taking the first device when devices is a string? eg devices could be "5". And when devices is an int, why is it taking the specific cuda device?. ie in the pl trainer, devices=3 would mean take the first 3 gpus?